### PR TITLE
version: 1.4.1

### PR DESCRIPTION
**Pull request by AllRWeak**

### CODE REVIEW

File: nova/core/roles/caddy/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        caddy_server_reverse_proxy_to_address: http://internal.example.com
+        caddy_use_tls: false
+```
+
+```yaml
+# Installing Caddy server that listens only on all ports with TLS enabled and reverse proxies to an internal server
+# Only responds to requests for www.example.com and www2.example.com
+- name: Installing Caddy...
+  ansible.builtin.include_role:
+    name: shared.roles.caddy
```
Comment (at3rva) [inline nova/core/roles/caddy/README.md:49]: Same here

File: nova/core/roles/caddy/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- name: Including Caddyfile with variables mode tasks...
+  ansible.builtin.include_tasks: caddyfile_with_vars.yml
+  when: caddy_template_caddyfile_with_vars
+
+- name: Including Caddyfile template tasks...
+  ansible.builtin.include_tasks: caddyfile_with_vars.yml
+  when: caddy_template_caddyfile
+
+- name: Composing Caddy for {{ inventory_hostname }}...
+  ansible.builtin.shell: docker compose -f {{ caddy_config_folder }}/docker-compose.yml up -d
```
Comment (at3rva) [inline nova/core/roles/caddy/tasks/main.yml:14]: Is it not better to use `community.docker.docker_compose`?

File: nova/core/roles/caddy/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Including Caddy installation tasks...
+  ansible.builtin.include_tasks: install.yml
+
+- name: Including Caddyfile with variables mode tasks...
+  ansible.builtin.include_tasks: caddyfile_with_vars.yml
+  when: caddy_template_caddyfile_with_vars
+
+- name: Including Caddyfile template tasks...
+  ansible.builtin.include_tasks: caddyfile_with_vars.yml
```
Comment (at3rva) [inline nova/core/roles/caddy/tasks/main.yml:10]: I think the file here is `caddy_template.yml`

**Review by at3rva — CHANGES_REQUESTED** [review]

File: nova/core/roles/caddy/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+- Depends on Docker and Docker Compose being installed on the host. Docker can be installed using the [nova.core.docker](https://github.com/novateams/nova.core/tree/main/nova/core/roles/caddy) role.
+
+## Example
+
+```yaml
+# Installing Caddy server that listens on all addresses and reverse proxies to an internal server
+- name: Installing Caddy...
+  ansible.builtin.include_role:
+    name: shared.roles.caddy
```
[the remark this answers is not in the document]
Comment (AllRWeak) [reply nova/core/roles/caddy/README.md]: Fixed

File: nova/core/roles/caddy/tasks/main.yml (hunk 2 above)
Comment (AllRWeak) [reply nova/core/roles/caddy/tasks/main.yml:14]: `community.docker.docker_compose` is using deprecated Docker API and no one has updated it. Currently the better way is to use shell that uses latest Docker Compose that comes with the compose plugin and is written in go.

File: nova/core/roles/caddy/tasks/main.yml (hunk 3 above)
Comment (AllRWeak) [reply nova/core/roles/caddy/tasks/main.yml]: Fixed

File: nova/core/roles/caddy/templates/caddy.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    ports:
+      - "80:80"
+      - "443:443"
+      - "443:443/udp"
+    volumes:
+      - /srv/caddy/etc:/etc/caddy # When only mounting Caddyfile the changes are not picked up inside the container
+      - /srv/caddy/data:/data/caddy
+      - /srv/caddy/config:/config/caddy
+      - /srv/caddy/logs:/srv/logs
+      - /srv/certs:/srv/certs
```
[the remark this answers is not in the document]
Comment (AllRWeak) [reply nova/core/roles/caddy/templates/caddy.yml]: Fixed

**Review by at3rva — APPROVED** [review]

**Pull request merged**